retrieval/transform_test.go

```diff
@@ -408,81 +432,98 @@ func TestSampleBuilder(t *testing.T) {
 			name: "summary",
 			metadata: promtest.MetadataMap{
 				"job1/instance1/metric1": &metadataEntry{Metric: "metric1", MetricType: textparse.MetricTypeSummary, ValueType: config.DOUBLE},
+				"job1/instance1/metric1_a_count": &metadataEntry{Metric: "metric1_a_count", MetricType: textparse.MetricTypeGauge, ValueType: config.DOUBLE},
 			},
 			series: seriesMap{
 				1: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1_sum"),
 				2: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1", "quantile", "0.5"),
 				3: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1_count"),
 				4: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1", "quantile", "0.9"),
+				// Add another series that only deviates by having an extra label. We must properly detect a new summary.
```

Comment:
Modeled the new tests inspired by the ones exposed by Histogram (replacing the old, simple test cases).